### PR TITLE
test: auto test

### PR DESCRIPTION
**Pull request by mokurin000**
In the existing `example/killmyargv.rs`, we have to check the correctness manually.
With `test-binary` and the helper binary, we could easily automate test.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,12 @@ log = "0.4.17"
 spdlog-rs = {features = ["log"], version = "0.3.12"}
 thiserror = "1.0.40"
 
+[dev-dependencies]
+sysinfo = { version = "0.30.5", default-features = false }
+test-binary = "3.0.2"
+base64 = "0.21.7"
+linereader = "0.4.0"
```
Comment (mokurin000) [inline Cargo.toml]: linereader is here for blocking read of ChildStdout, thus we wait for it completes the `killmyargv::set` operation

File: testbin/set_cmdline_from_stdin/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    let engine = GeneralPurpose::new(&alphabet, config);
+
+    let kill_my_argv = KillMyArgv::new()?;
+
+    for next_cmd_line in stdin.lines() {
+        let cmd_line = next_cmd_line?;
+        let cmd_line = engine.decode(&cmd_line)?;
+        kill_my_argv.set(&cmd_line);
+
+        println!("set done");
```
Comment (mokurin000) [inline testbin/set_cmdline_from_stdin/src/main.rs]: with one line output, we say this time we had completed set operation
do not output new lines here

**Comment by unknowndevQwQ** [comment]
建议把两个 commit 压缩为一个，或者我这里尝试用 squash 合并，但是这样我不能控制 commit 描述

**Comment by mokurin000** [comment]
> 建议把两个 commit 压缩为一个，或者我这里尝试用 squash 合并，但是这样我不能控制 commit 描述

没有压在一起是因为前者更方便并发测试多项功能
在Cargo.toml我们无法设置test thread

**Pull request merged**